lib/post_destroyer.rb

```diff
@@ -166,7 +176,7 @@ def perform_delete
     # All posts in the topic must be force deleted if the first is force
     # deleted (except @post which is destroyed by current instance).
     if @topic && @post.is_first_post? && permanent?
-      @topic.ordered_posts.with_deleted.reverse_order.find_each do |post|
+      @topic.posts.with_deleted.find_each do |post|
```

Comment:
When using `#find_each`, any order scope will be overridden, because the batching feature uses ordering on the pkey for its own functionality. Removing the ordering here silences a warning we'd otherwise get that the order won't apply.